### DISCUSSION TITLE
aria2 1.37.0

### DISCUSSION THREAD

**Discussion by tatsuhiro-t** [aria2/aria2#2136] [Announcements]
## What's Changed
* Fix header in --http-accept-gzip documentation by @richardnias in https://github.com/aria2/aria2/pull/1824
* Fix typo in documentation, --help text by @h3xx in https://github.com/aria2/aria2/pull/1877
* Update aria2c.rst by @ITriskTI in https://github.com/aria2/aria2/pull/1902
* Allow empty dist name in bencode which is needed for hybrid torrent by @tatsuhiro-t in https://github.com/aria2/aria2/pull/1905
* Better entropy for getRandomBytes by @tatsuhiro-t in https://github.com/aria2/aria2/pull/1907
* Deal with missing nproc on macos by @tatsuhiro-t in https://github.com/aria2/aria2/pull/1924
* Bump actions by @tatsuhiro-t in https://github.com/aria2/aria2/pull/1923
* Bump workflow ubuntu by @tatsuhiro-t in https://github.com/aria2/aria2/pull/1929
* Add missing <array> include to WinTLSSession.h by @hmartinez82 in https://github.com/aria2/aria2/pull/1943
* Fix undefined behavior/crash in GZipEncoder by @himikof in https://github.com/aria2/aria2/pull/1970
* Fix Metalink4 parsing with foreign namespaces by @himikof in https://github.com/aria2/aria2/pull/1969
* fix wrong `dht.dat` binary file structure in docs by @yixinBC in https://github.com/aria2/aria2/pull/2068
* Increase ByteArrayDiskWriter maximum size by @kelson42 in https://github.com/aria2/aria2/pull/1893
* Minor grammar improvements by @nkh0472 in https://github.com/aria2/aria2/pull/1804
* Fix static link failure against libssh2 by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2088
* Update Dockerfile.mingw by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2089
* Prefer random number generator from crypto libraries by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2091
* android(ndk r23) has timegm by @FH0 in https://github.com/aria2/aria2/pull/1822
* Add Dockerfile.android by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2092
* Remove deprecated std::unary_function and std::binary_function by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2093
* Update wslay by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2094
* Fix test errors with ubsan by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2095
* ci: Build with gnutls by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2096
* ci: Build mingw image by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2097
* Revert "ci: Build mingw image" by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2098
* Fix overflow by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2099
* Make releases with docker by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2100
* Dockerfile.mingw: Parallel build by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2101
* Dockerfile.android: Add dpkg-dev for dpkg-architecture by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2102
* Dockerfile.mingw: Remove deprecated libssh2 configure flags by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2103
* Dockerfile.mingw: Update how to get aria2c.exe from a container by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2104
* Update sphinx_rtd_theme by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2105
* Static check fix by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2106
* Do not close stdout and stderr by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2107
* Avoid non-nil argument errors by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2108
* Logger: Fix format string overflow in writeHeader() by @egorenar in https://github.com/aria2/aria2/pull/2117
* ci: Bump gcc and clang by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2125
* Do not require strict C++ mode and update ax_cxx_compile_stdcxx.m4 by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2126
* Cap infoHashLength in .aria2 file by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2127
* Fix non bt build error by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2129
* Various documenation fixes and rewords  by @h3xx in https://github.com/aria2/aria2/pull/1982
* Change 'meta data' to 'metadata' by @h3xx in https://github.com/aria2/aria2/pull/1983
* Dockerfile: Bump c-ares to 1.21.0 by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2132
* Dockerfile.mingw: Downgrade c-ares to 1.19.1 by @tatsuhiro-t in https://github.com/aria2/aria2/pull/2135

## New Contributors
* @richardnias made their first contribution in https://github.com/aria2/aria2/pull/1824
* @h3xx made their first contribution in https://github.com/aria2/aria2/pull/1877
* @hmartinez82 made their first contribution in https://github.com/aria2/aria2/pull/1943
* @himikof made their first contribution in https://github.com/aria2/aria2/pull/1970
* @yixinBC made their first contribution in https://github.com/aria2/aria2/pull/2068
* @kelson42 made their first contribution in https://github.com/aria2/aria2/pull/1893
* @nkh0472 made their first contribution in https://github.com/aria2/aria2/pull/1804
* @FH0 made their first contribution in https://github.com/aria2/aria2/pull/1822
* @egorenar made their first contribution in https://github.com/aria2/aria2/pull/2117

**Full Changelog**: https://github.com/aria2/aria2/compare/release-1.36.0...release-1.37.0

<hr /><em>This discussion was created from the release <a href='https://github.com/aria2/aria2/releases/tag/release-1.37.0'>aria2 1.37.0</a>.</em>

**Comment by fenopa** [comment]
I would expect a fix for this before a new release: https://github.com/aria2/aria2/issues/1118

**Comment by andikorol** [comment]
Thanks for the effort, great work!
When we built the 1.36 on macos x64 and arm64 without torrent and metalink support, the binaries were around 3 mb. The ones from 1.37 are around 9 mb. 
Any idea what the reason for that is? - Thanks!

**Comment by randomComitter** [comment]
Will mac binaries be also delivered for this release?

**Comment by AncientTree** [comment]
Is there any distribution for Mac OS?

**Reply by hrxn** [reply]
@AncientTree Did you try it with homebrew?

**Reply by h3xx** [reply]
*30 second google search later* Yes. 
```sh
brew install aria2
```
See: https://formulae.brew.sh/formula/aria2

**Reply by AncientTree** [reply]
> @AncientTree Did you try it with homebrew?

Thanks for your advice, that's helpful.

**Comment by Touby22** [comment]
Thanks for your great Tool, hope you soon update this tool and consider:
- reduce tool total size
- add progress bar like this :
[https://i.sstatic.net/G7xH3.png](url)
<img width="499" height="322" alt="G7xH3" src="https://github.com/user-attachments/assets/2dbff6f3-060c-4ffb-bcc3-94e86f6d43db" />
